holmes/plugins/toolsets/kubernetes_logs.py

```diff
@@ -0,0 +1,284 @@
+
+            query_params["timestamps"] = filter_by_timestamps
+
+            if container_name:
+                query_params["container"] = container_name
+
+            logs = self._core_v1_api.read_namespaced_pod_log(**query_params)
+
+            if logs:
+                # Split logs by newline but filter out empty lines
+                logs = [line for line in logs.strip().split("\n") if line]
+
+                if filter_by_timestamps:
+                    start, end = process_timestamps_to_int(
+                        start=params.start_time,
+                        end=params.end_time,
+                        default_time_span_seconds=3600,
+                    )
+                    return filter_log_lines_by_timestamp_and_strip_prefix(
+                        logs=logs, start_unix_timestamp=start, end_unix_timestamp=end
+                    )
+                else:
+                    return logs
+
+            return []
+        except ApiException as e:
+            if e.status == 400 and "previous terminated container" in str(e).lower():
+                return []
+            elif e.status == 404:
+                return []
+            else:
+                logging.warning(
+                    f"API error fetching logs. params={query_params}. Error: {str(e)}"
+                )
+                raise e
+        except Exception as e:
+            logging.error(
+                f"Error fetching logs. params={query_params}. Error: {str(e)}"
+            )
+            raise
+
+    def _filter_logs(self, logs: List[str], pattern: str) -> List[str]:
+        """Filter logs by regex pattern"""
+        try:
+            regex = re.compile(pattern)
+            return [log for log in logs if regex.search(log)]
+        except re.error as e:
+            logging.warning(f"Invalid regex pattern: {pattern}, error: {str(e)}")
+            return logs  # Return unfiltered logs on regex error
+
+    def _format_logs(self, logs: List[str]) -> str:
+        """Format logs for output"""
+        if not logs:
+            return "No logs found"
+
+        # Simple join without line numbers to match kubectl logs format
+        return "\n".join([log for log in logs if log.strip()])
+
+
+def filter_log_lines_by_timestamp_and_strip_prefix(
+    logs: List[str], start_unix_timestamp: int, end_unix_timestamp: int
+) -> List[str]:
+    """
+    Filters log lines based on their leading ISO 8601 timestamp, keeping lines
+    within the specified start and end Unix timestamps (inclusive). Returns the
+    filtered lines *without* the leading timestamp prefix.
+    """
+    # match ISO 8601 format (YYYY-MM-DDTHH:MM:SS[.fffffffff]Z)
+    timestamp_pattern = re.compile(
+        r"^(?P<ts>\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}(?:\.\d+)?(?:Z|[+-]\d{2}:\d{2}))"
+    )
+    filtered_lines_content: List[str] = []
+
+    for line in logs:
+        # Non-string entries are passed through as-is
+        if not isinstance(line, str):
```

Comment:
What is this case? can you add also a commnt to document it?

Reply:
It's defensive code. should actually be `filtered_lines_content.append(str(line))`. Fixed